query-algebrizer/src/types.rs

```diff
@@ -691,6 +693,22 @@ impl Debug for EmptyBecause {
     }
 }
 
+
+/// A `FindQuery` represents a valid query to the query algebrizer.
```

Comment:
I personally would like "ValidFindQuery" a bit more - or perhaps the more specific "AlgebrizableFindQuery". To me the current naming is very vague in context of the split from ParsedFindQuery.

Reply:
Especially given that this is conceptually a "sub type" of the more generic ParsedFindQuery, right? In a sense that it encompasses a smaller number of possibilities.

Reply:
I don't want to change the name `FindQuery`, so I'll change `ParsedFindQuery` to `ParsedQuery`.